js-pkg/client/src/encryption.ts

```diff
@@ -0,0 +1,51 @@
+function arrayBufferToBase64(buffer: ArrayBuffer): string {
+  const bytes = new Uint8Array(buffer)
+  let binary = ''
+  for (let i = 0; i < bytes.byteLength; i++) {
+    binary += String.fromCharCode(bytes[i])
+  }
```

Comment:
Unfortunately not, `TextDecoder` does codepoint conversions rather than base conversions. Seems like `atob` and `btoa` are the best-supported approach, as gross as they are. `UInt8Array` has [fromBase64](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint8Array/fromBase64), but the support is abysmal.

Reply:
I wasn't able to make it work with `TextDecoder` but `String.fromCharCode` can take a splat which is a bit nicer than the loop.